File: Solutions/0202- Happy Number/Solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();

        while (n != 1 && !set.contains(n)) {
            set.add(n);
            int temp = 0;
            while (n != 0) {
                temp += (int) Math.pow((n % 10),2);
                n /= 10;
            }
            n = temp;
        }

        return n == 1;

    }
}

File: Solutions/0202- Happy Number/Solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        my_set = set()

        while n != 1 and n not in my_set:
            my_set.add(n)
            temp = 0
            while (n != 0):
                temp += pow((n % 10), 2)
                n //= 10
            n = temp

        return n == 1
        